tests/test_metadata_serialization.py

```diff
@@ -360,6 +360,43 @@ def test_delegation_serialization(self, test_case_data: str):
         delegation = Delegations.from_dict(copy.deepcopy(case_dict))
         self.assertDictEqual(case_dict, delegation.to_dict())
 
+    valid_delegations: utils.DataSet = {
```

Comment:
Can you please use the dataset defined at line 321 https://github.com/theupdateframework/python-tuf/blob/acb201d6cc685c2560f1000faf47e75db73df7fb/tests/test_metadata_serialization.py#L321
and the test function defined in 336 https://github.com/theupdateframework/python-tuf/blob/acb201d6cc685c2560f1000faf47e75db73df7fb/tests/test_metadata_serialization.py#L336
instead of defining a `valid_delegations` and a new test function?

We aim to group all valid tests related to a specific class in one dataset and test function.
Also, the name `valid_delegations` is not correct.
Those are actually cases of `invalid_delegations` as when you call `Delegations.from_dict(copy.deepcopy(case_dict))` it causes `ValueError` as you catch it inside your the test function.

Reply:
I have a concern about how to make sure that the ``ValueError`` is the correct unit we want.
I mean, we test the ``ValueError`` raised, but we don't check if the message comes from the part we want to cover.
Unless we don't want to go deep in this level, it is ok.

Reply:
I think this is a more generic question. 
I think for sure we want to have all invalid tests for `Delegations` in one single place, but I understand what you mean.
Maybe we can check the error message and type inside the with?
What do you think @jku?

Reply:
Yes, indeed. If agreed in checking the type and error message, we can address it in another issue.